src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -300,7 +303,8 @@ public bool IsConnected
             this.onDisconnected = onDisconnected;
             this.onSendingMessage = onSendingMessage;
 
-            this.asyncQueue = new AsyncQueue<Payload>(this.SendMessageHandledAsync);
+            string dequeuerName = $"{nameof(NetworkPeer)}-{nameof(this.asyncPayloadsQueue)}-{this.PeerEndPoint.ToString()}";
```

Comment:
you can use peer id instead (but maybe endpoint is not a bad idea then we can immediately connect the failing peer)

Reply:
yes it was to have an easier way to see the correlation in console output